lib/Widgets/SettingsSidebar/SettingsSidebarRow.vala

```diff
@@ -0,0 +1,124 @@
+/*
+* Copyright (c) 2017 elementary LLC. (https://elementary.io)
+*
+* This program is free software: you can redistribute it and/or modify
+* it under the terms of the GNU Lesser General Public License as published by
+* the Free Software Foundation, either version 2.1 of the License, or
+* (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU Library General Public License for more details.
+*
+* You should have received a copy of the GNU Lesser General Public
+* License along with this program; if not, write to the
+* Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+* Boston, MA 02110-1301 USA
+*/
+
+private class Granite.SettingsSidebarRow : Gtk.ListBoxRow {
+    public SettingsPage.StatusType status_type {
+        set {
+            switch (value) {
+                case SettingsPage.StatusType.ERROR:
+                    status_icon.icon_name = "user-busy";
+                    break;
+                case SettingsPage.StatusType.OFFLINE:
+                    status_icon.icon_name = "user-offline";
+                    break;
+                case SettingsPage.StatusType.SUCCESS:
+                    status_icon.icon_name = "user-available";
+                    break;
+                case SettingsPage.StatusType.WARNING:
+                    status_icon.icon_name = "user-away";
+                    break;
+            }
+        }
+    }
+
+    public Gtk.Widget display_widget { get; construct; }
+
+    public string? header { get; set; }
+
+    public string icon_name {
+        get {
+            return _icon_name;
+        }
+        set {
+            _icon_name = value;
+            ((Gtk.Image) display_widget).icon_name = value;
```

Comment:
You make the `display_widget` be an arbitrary type of `Gtk.Widget`, and here you cast it to `Gtk.Image` which cannot be done like that. You have to check if the widget is indeed a `Gtk.Image` here.